arm/resources/subscriptions/models.go

```diff
@@ -24,6 +24,35 @@ import (
 	"net/http"
 )
 
+// SpendingLimit enumerates the values for spending limit.
+type SpendingLimit string
+
+const (
+	// CurrentPeriodOff specifies the current period off state for spending
+	// limit.
+	CurrentPeriodOff SpendingLimit = "CurrentPeriodOff"
+	// Off specifies the off state for spending limit.
+	Off SpendingLimit = "Off"
+	// On specifies the on state for spending limit.
+	On SpendingLimit = "On"
+)
+
+// tate enumerates the values for tate.
+type tate string
```

Comment:
I revert API version to 2015-.. because I am not sure how we are going to handle this case. It is not a bug, it is the way we handle stuttering types.

We remove package name from type names to avoid stuttering type golint error.
For example, if there is a swagger for API ABC and it has type ABCAccount, then to use this type in go `ABC.ABCAccount` which gives golint error. To avoid this, we trim the package name in typenames. `ABCAccount` -> `Account`.

In this case, `tate`..
package: `subscriptions`
Enum type: `SubscriptionState` -> after trimming `subscriptions`, it becomes `tate`.